kafka/src/main/java/io/micronaut/configuration/kafka/processor/ConsumerInfo.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.kafka.processor;
+
+import io.micronaut.configuration.kafka.KafkaMessage;
+import io.micronaut.configuration.kafka.annotation.ErrorStrategy;
+import io.micronaut.configuration.kafka.annotation.ErrorStrategyValue;
+import io.micronaut.configuration.kafka.annotation.KafkaListener;
+import io.micronaut.configuration.kafka.annotation.OffsetStrategy;
+import io.micronaut.configuration.kafka.seek.KafkaSeekOperations;
+import io.micronaut.core.annotation.*;
+import io.micronaut.core.reflect.ReflectionUtils;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.util.ArrayUtils;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.inject.ExecutableMethod;
+import io.micronaut.messaging.Acknowledgement;
+import io.micronaut.messaging.annotation.SendTo;
+import io.micronaut.messaging.exceptions.MessagingSystemException;
+import org.apache.kafka.clients.consumer.Consumer;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Internal consumer info.
+ */
+@Internal
+final class ConsumerInfo {
```

Comment:
What is the point of extracting all the values into a new class?

Reply:
The point is twofold:

- Separate pieces of data that can be evaluated just once into an immutable class.
- Simplify `ConsumerState` which used to have 25 fields. Now `ConsumerState` is in charge of just "state" (ie. mutable fields).

Reply:
Ok, you might want to make it a record

Reply:
Actually `ConsumerInfo` started off as a record. It proved impractical, so I had to turn it into a regular final class.